api/grpcserver/gateway_service.go

```diff
@@ -0,0 +1,49 @@
+package grpcserver
+
+import (
+	"github.com/spacemeshos/go-spacemesh/activation"
+
+	pb "github.com/spacemeshos/api/release/go/spacemesh/v1"
+	"github.com/spacemeshos/go-spacemesh/api"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"golang.org/x/net/context"
+	"google.golang.org/genproto/googleapis/rpc/code"
+	rpcstatus "google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// GatewayService exposes transaction data, and a submit tx endpoint
+type GatewayService struct {
+	Network api.NetworkAPI // P2P Swarm
+}
+
+// RegisterService registers this service with a grpc server instance
+func (s GatewayService) RegisterService(server *Server) {
+	pb.RegisterGatewayServiceServer(server.GrpcServer, s)
+}
+
+// NewGatewayService creates a new grpc service using config data.
+func NewGatewayService(net api.NetworkAPI) *GatewayService {
+	return &GatewayService{
+		Network: net,
+	}
+}
+
+// BroadcastPoet accepts a binary poet message to broadcast to the network
+func (s GatewayService) BroadcastPoet(_ context.Context, in *pb.BroadcastPoetRequest) (*pb.BroadcastPoetResponse, error) {
+	log.Info("GRPC GatewayService.BroadcastPoet")
```

Comment:
don't we use a custom logger for the GRPC? if not I think we should

Reply:
I agree, it's a work in progress, see #2086